Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -0,0 +1,1765 @@
+
+    recovered_items = set()
+    for item in items_to_recover:
+        recovered_items.add(item)
+    if len(recovered_items) != len(message_ids):
+        missing_items = set(message_ids).difference(recovered_items)
+        raise Exception(
+            f"Some message ids are missing in recoverable items directory: {missing_items}"
+        )
+
+    for item in recovered_items:
+        item.move(target_folder)
+        recovered_messages.append(
+            {ITEM_ID: item.id, MESSAGE_ID: item.message_id, ACTION: "recovered"}
+        )
+
+    readable_output = tableToMarkdown("Recovered messages", recovered_messages)
+    output = {CONTEXT_UPDATE_EWS_ITEM: recovered_messages}
+    return readable_output, output, recovered_messages
+
+
+def get_contacts(client: EWSClient, limit, target_mailbox=None):
+    def parse_physical_address(address):
+        result = {}
+        for attr in ["city", "country", "label", "state", "street", "zipcode"]:
+            result[attr] = getattr(address, attr, None)
+        return result
+
+    def parse_phone_number(phone_number):
+        result = {}
+        for attr in ["label", "phone_number"]:
+            result[attr] = getattr(phone_number, attr, None)
+        return result
+
+    def parse_contact(contact):
+        contact_dict = dict(
+            (k, v if not isinstance(v, EWSDateTime) else v.ewsformat())
+            for k, v in list(contact._field_vals())
+            if isinstance(v, str) or isinstance(v, EWSDateTime)
+        )
+        if isinstance(contact, Contact) and contact.physical_addresses:
+            contact_dict["physical_addresses"] = list(
+                map(parse_physical_address, contact.physical_addresses)
+            )
+        if isinstance(contact, Contact) and contact.phone_numbers:
+            contact_dict["phone_numbers"] = list(
+                map(parse_phone_number, contact.phone_numbers)
+            )
+        if (
+            isinstance(contact, Contact)
+            and contact.email_addresses
+            and len(contact.email_addresses) > 0
+        ):
+            contact_dict["emailAddresses"] = [x.email for x in contact.email_addresses]
+        contact_dict = keys_to_camel_case(contact_dict)
+        contact_dict = dict((k, v) for k, v in list(contact_dict.items()) if v)
+        contact_dict.pop("mimeContent", None)
+        contact_dict["originMailbox"] = target_mailbox
+        return contact_dict
+
+    account = client.get_account(target_mailbox)
+    contacts = []
+
+    for contact in account.contacts.all()[: int(limit)]:  # pylint: disable=E1101
+        contacts.append(parse_contact(contact))
+    readable_output = tableToMarkdown(f"Email contacts for {target_mailbox}", contacts)
+    output = {"Account.Email(val.Address == obj.originMailbox).EwsContacts": contacts}
+    return readable_output, output, contacts
+
+
+def create_folder(client: EWSClient, new_folder_name, folder_path, target_mailbox=None):
```

Comment:
`target_mailbox` is not used here

Reply:
nice catch, this was missed in the migration - it's an optional value. Added relevant code.